library/bignum.c

```diff
@@ -674,14 +674,68 @@ int mbedtls_mpi_write_file( const char *p, const mbedtls_mpi *X, int radix, FILE
 }
 #endif /* MBEDTLS_FS_IO */
 
+
+/* Convert a big-endian byte array aligned to the size of mbedtls_mpi_uint
+ * into the storage form used by mbedtls_mpi. */
+static void mpi_bigendian_to_host( mbedtls_mpi_uint * const p, size_t limbs )
+{
+    size_t i;
+
+    unsigned char *cur_byte_left;
+    unsigned char *cur_byte_right;
+
+    mbedtls_mpi_uint *cur_limb_left;
+    mbedtls_mpi_uint *cur_limb_right;
+
+    mbedtls_mpi_uint tmp_left, tmp_right;
+
+    if( limbs == 0 )
+        return;
+
+    /*
+     * Traverse limbs and
+     * - adapt byte-order in each limb
+     * - swap the limbs themselves.
+     * For that, simultaneously traverse the limbs from left to right
+     * and from right to left, as long as the left index is not bigger
+     * than the right index (it's not a problem if limbs is odd and the
+     * indices coincide in the last iteration).
+     */
+
+    for( cur_limb_left = p, cur_limb_right = p + ( limbs - 1 );
+         cur_limb_left <= cur_limb_right;
+         cur_limb_left++, cur_limb_right-- )
```

Comment:
@yanesca Thanks for your review, and apologies for the late response!
> ... but I think that in general it is good practice to avoid pointer arithmetic unless there is a compelling reason to use it. 

I'm not sure I agree here. Why would you avoid pointer arithmetic, especially if it's incremental / decremental one like here? Also, I'd expect index arithmetic to make code slightly larger as you need two arithmetic operations -- the index arithmetic and calculating the shifted pointer -- vs. only one -- shifting the pointer -- for pointer arithmetic.

I just tested this: On x86, compiling the current code with `-Os` on GCC produces 84 Bytes of assembly, while using index arithmetic would yield 97 Bytes. For Arm the difference is 74 Bytes vs. 82 Bytes. Certainly not a big difference, but maybe worth considering when it comes to a general decision of what idiom to use.

Anyway, I spent a bit of time making use of compiler specific byte swapping functions, which allowed to reduce the code size to 46 Bytes on x86 and to 36 Bytes on Arm. Also, I think the code got more readable through the change. So hopefully we can go with that?